docs/runtime-contract.md

```diff
@@ -181,51 +157,39 @@ for purposes of scaling CPU and removing idle containers.
 
 #### Protocols and Ports
 
-The container MUST accept HTTP/1.1 requests from the environment. The
-environment
-[SHOULD offer an HTTP/2.0 upgrade option](https://http2.github.io/http2-spec/#discover-http)
```

Comment:
I would have liked to keep the HTTP/2.0 upgrade option suggestion, though I can understand removing it or downgrading it to MAY since we're not able to make Envoy do this yet.

Reply:
So question on this for you. We had previous defined this as 3 modes:

(empty): HTTP/1.1 with automatic upgrades
"h2c": HTTP/2 only
"http1": HTTP/1.1 only

Our support today looks like:

(empty): HTTP/1.1 only
"h2c": HTTP/2 only
"http1": HTTP/1.1 only

This makes (empty) == "http1" in todays world. 

I have concerns with making (empty) automatically upgrade to HTTP/2.0 in the future and leaving "http1" as the option with no upgrade. This seems like it has the potential to 1) potentially be breaking 2) surprising. Our considerations could change by the time we get around to implementing it so I am tempted to keep the automatic upgrade out for now.

Reply:
I opened https://github.com/knative/serving/pull/4286 so we can get this merged in separate from the rest of the changes here. Lets continue this conversation there.